### DISCUSSION TITLE
How to link to CPA codelist

### DISCUSSION THREAD

**Discussion by JohannaOtt** [INSPIRE-MIF/helpdesk#130] [Q&A]
Hi everybody,

In ActivityComplex.input and ActivityComplex.output, the codelist https://inspire.ec.europa.eu/codelist/InputOutputValue should be used. It points to the external codelist http://ec.europa.eu/eurostat/ramon/nomenclatures/index.cfm?TargetUrl=LST_NOM_DTL&StrNom=CPA_2008&StrLanguageCode=EN&IntPcKey=&StrLayoutCode=HIERARCHIC

When navigating to one code of this codelist the URL is for example "https://ec.europa.eu/eurostat/ramon/nomenclatures/index.cfm?TargetUrl=DSP_NOM_DTL_VIEW&StrNom=CPA_2008&StrLanguageCode=EN&IntPcKey=20997638&IntKey=20997741&StrLayoutCode=HIERARCHIC&IntCurrentPage=1".

Is it expected to use that kind of value for input.href or should the value be provided in a different format?

All the best

Johanna

**Comment by heidivanparys** [comment]
I am not familiar with the [legislation regarding CPA](https://eur-lex.europa.eu/eli/reg/2008/451/2019-07-26), I'm afraid.

I looked at the following links:

- [legislation regarding CPA, consolidated text](https://eur-lex.europa.eu/eli/reg/2008/451/2019-07-26)
- https://ec.europa.eu/eurostat/web/cpa/cpa_2.1
- https://eur-lex.europa.eu/eli/reg/2014/1209/oj

According to https://ec.europa.eu/eurostat/web/cpa/cpa_2.1:

> In November 2012, an update of the Classification of Products by Activity (CPA) was launched. The Regulation establishing CPA version 2.1 was adopted in October 2014, entering into force 1 January 2015.

So if I understand it correct, the regulation establishing CPA version 2.1 applies since 1 January 2015, and CPA 2008 is superseded.

CPA 2.1 is available [here](https://op.europa.eu/en/web/eu-vocabularies/dataset/-/resource?uri=http://publications.europa.eu/resource/dataset/statistical-classification), see also http://publications.europa.eu/resource/authority/cpa21, and a specific value can be referred with a URI, e.g. http://publications.europa.eu/resource/authority/cpa21/01.11.11: `xlink:href=http://publications.europa.eu/resource/authority/cpa21/01.11.11`.

@MarcoMinghini Additional feedback would be needed here. Is what I'm saying correct? If yes, a change for the INSPIRE registry will have to be submitted, to update the external reference link at https://inspire.ec.europa.eu/document/StatProd.

**Comment by heidivanparys** [comment]
@JohannaOtt Have you had the chance to look at the answer provided? Has your question been answered?

**Comment by JohannaOtt** [comment]
@heidivanparys Thank you for your input. I am ok with using the URLs from the codelist you provided. My assumption was that we are still awaiting feedback from @MarcoMinghini on this topic before moving on.

**Comment by jescriu** [comment]
Dear @heidivanparys,

Thank you for you deep analysis, specially for noticing the potential obsolescence of the 'Statistical Classification of Products by Activity in the European Community, 2008 version' code list.

In my personal view, use of externally managed code lists should be minimised as much as possible in the INSPIRE Registry - Just because: 

1. If there are any changes to the code list, these changes would probably remain unnoticed to the INSPIRE Registry Control Body, which is not a desirable situation. This becomes more important when the changes to the external code list are not controlled, i.e. when the external code list are not managed in a formal register (such as the INSPIRE Registry).
2. When the code list are managed in the INSPIRE Registry, their values might be appropriately referenced - Which is the feature @JohannaOtt is currently missing.

Having this said, my suggestion would be to:

1.- Prepare a first change request in the INSPIRE Registry, to add the full set of values corresponding to the 'Statistical Classification of Products by Activity in the European Community, 2008 version' (namely CPA version 2008) to the already existing an approved https://inspire.ec.europa.eu/codelist/ProductCPAValue code list. 

Although this version of the code list seems to be outdated, many datasets may still reference them. 

See NOTE (*) below.

2.- Prepare a second change request in the INSPIRE Registry, to add the 'Statistical Classification of Products by Activity version 2.1' code list (namely CPA version 2.1) as a new version of the previous one.

This way it would be available to be referenced from datasets already compliant to the COMMISSION REGULATION (EU) No 1209/2014, of 29 October 2014.

Although it would not be my preferred option, I cannot see major issue in keeping this new version of the code list as externally governed - since it is already published in the system of the Publications Office of the European Union and, as you @heidivanparys pointed, its values can be appropriately referenced.

* NOTE: To be further analysed if CPA 2008 version should indeed be superseded, or should stay as a 'Valid' register - Backwards compatibility from CPA version 2.1 to CPA version 2008 should be taken into account. Additionally, EuroSTAT expertise could be required for taking a final decision. 

If these change requests are finally proposed, they should be transferred to the INSPIRE Registry Helpdesk (https://github.com/INSPIRE-MIF/helpdesk-registry).

Anyway, here I will ask @hernlor to provide her expert advice.

All the best,
Jordi
